Sources/NIO/ByteBuffer-core.swift

```diff
@@ -640,8 +640,8 @@ public struct ByteBuffer {
             // the slice's begin is past the maximum supported slice begin value (16 MiB) so the only option we have
```

Comment:
There's almost no correct way to use the `sliceStartIndex` variable. I wonder if it would be nicer to eliminate it and put it straight into the condition. Ie. 

    guard self._slice.lowerBound + index <= ByteBuffer.Slice.maxSupportedLowerBound else

Reply:
It is also used after the guard condition in L651.

Reply:
fair